### DISCUSSION TITLE
How do I tell DetailsList that I want a particular row to be selected?

### DISCUSSION THREAD

**Discussion by najmera30** [microsoft/fluentui#23289] [Q&A]
The selection for the row is based on an id matching basis. If the ids match (i.e if true), I want to set the row as selected. If not, then not-selected. How do I achieve this in a functional component using 'useState' and 'useEffect'. Also, I want to use setIndex instead of setKey. Any examples of custom selection?

**Comment by spmonahan** [comment]
Hi @najmera30,

I'm not sure I understand exactly what you're asking. Can you share a [Codepen](http://aka.ms/fluentpen) with what you've tried to get this working so I can get a better understanding of what you're trying to accomplish?
